ballerina/channel_csv_io.bal

```diff
@@ -40,30 +40,77 @@ Error {
     return (check getReadableCSVChannel(readableChannel, 0)).csvStream();
 }
 
-isolated function channelWriteCsv(WritableChannel writableChannel, string[][]|map<anydata>[] content) returns Error? {
-    WritableCSVChannel csvChannel = check getWritableCSVChannel(writableChannel);
+isolated function channelWriteCsv(string path, FileWriteOption option, string[][]|map<anydata>[] content) returns Error? {
+    
+    WritableCSVChannel csvChannel;
     if content is string[][] {
+        csvChannel = check getWritableCSVChannel(check openWritableCsvFile(path, option = option));
         foreach string[] r in content {
             Error? writeResult = csvChannel.write(r);
             if writeResult is Error {
                 check csvChannel.close();
                 return writeResult;
             }
         }
+        check csvChannel.close();
     } else if content is map<anydata>[] {
+        string[] headers = [];
+        if (option == OVERWRITE) {
+            headers = content[0].keys();
```

Comment:
Can there by indexoutof bounds here? What if the file does not contain any records?

Reply:
Updated the code to check for empty inputs in the [706129c](https://github.com/ballerina-platform/module-ballerina-io/pull/827/commits/706129c42b92d795da38857133f1af12c89b6c3b). This won't be a problem for string[][] and we only have to check for Record[]

Reply:
Again the same question, even in overwite scenario do we expect a csv file with headers?

Reply:
No. This line is used to get the headers from the user given content i.e. map<>[]|string[][]. And it has to be obtained in both OVERWRITE and APPEND scenarios as we have to either write or check the headers.